src/body.rs

```diff
@@ -474,13 +474,41 @@ impl Body {
     }
 
     /// Returns the mime type of this Body.
-    pub fn mime(&self) -> &Mime {
-        &self.mime
+    pub fn mime(&self) -> Option<&Mime> {
+        self.mime.as_ref()
     }
 
     /// Sets the mime type of this Body.
+    ///
+    /// # Examples
+    /// ```
+    /// use http_types::Body;
+    /// use http_types::mime::self;
+    ///
+    /// let mut body = Body::default();
+    /// body.set_mime(Some(mime::CSS));
+    /// assert_eq!(*body.mime(), mime::CSS);
+    /// ```
     pub fn set_mime(&mut self, mime: impl Into<Mime>) {
-        self.mime = mime.into();
+        self.mime = Some(mime.into());
+    }
+
+    /// Unsets the mime type of this Body.
+    ///
+    /// # Examples
+    /// ```
+    /// use http_types::Body;
+    /// use http_types::mime;
+    ///
+    /// let mut body = Body::default();
+    /// assert!(body.mime().is_some());
+    ///
+    /// body.unset_mime();
+    /// assert!(body.mime().is_none());
+    /// ```
+    ///
+    pub fn unset_mime(&mut self) {
+        self.mime = None
```

Comment:
I was hoping we could let `set_mime` take an `Option`.

Reply:
If we want to accept an `Option<impl Into<Mime>>`, all uses of `None` in argument position would have to be disambiguated for the compiler:
```rust
body.set_mime(Option::<Mime>::None);
```
This (I think) doesn't look great, and if the user does it wrong, the compiler doesn't give a great error message (because it can't infer what might be a reasonable type for T):

```rust
error[E0283]: type annotations needed
 --> src/body.rs:489:6
  |
8 | body.set_mime(None);
  |      ^^^^^^^^ cannot infer type for type parameter `T` declared on the associated function `set_mime`
  |
  = note: cannot satisfy `_: Into<Mime>`

error: aborting due to previous error

For more information about this error, try `rustc --explain E0283`.
```

If we want the generic Into<Mime>, I think the best alternative would be an unset function. An alternative (still keeping Into would be changing the signature to

```rust
pub fn set_mime(&mut self, mime: impl Into<Option<Mime>>);
```

but I don't like this for
1. A performance reason: It introduces an otherwise unnecessary branch in set_mime (a bit of code I believe could be called quite often), and is unlikely to be inlined as it crosses (at least) one create boundary (this would have to be benchmarked though, and could probably be solved with `#[inline(always)]`),
2. A ecosystem breakage reason: It would break any custom Mime implementation both in binary and library code for (in my opinion) not much gain, and
3. A style reason: I think it seems weird implementing `Into` for an `Option`, and using the error type of `TryInto` as unsetting the mime could lead to bad debugging experiences (tracing through calls until you find something just eating the error).

I'll add that I think it would be a fine option to change the signature to:
```rust
pub fn set_mime(&mut self, mime: Option<Mime>)
```
This would solve all problems with disambiguation, and ensures that "bad" Into implementations panic in the callers code, but I'm not sure how much developer experience would suffer (or if at all).

Reply:
> I'll add that I think it would be a fine option to change the signature to:
> 
> ```rust
> pub fn set_mime(&mut self, mime: Option<Mime>)
> ```
> 
> This would solve all problems with disambiguation, and ensures that "bad" Into implementations panic in the callers code, but I'm not sure how much developer experience would suffer (or if at all).

That indeed seems like the better option. Happy for us to with that!